### PR TITLE
Making proto write functions more flexible then they were previously

### PR DESCRIPTION
**Pull request by AsherGlick**
Proto write functions now no longer return their data, instead they get to directly write their data to the proto itself. There are two variants for scalar and non-scalar, these apply 1-1 to the set() and set_allocated() functions and were pretty unavoidable.

### CODE REVIEW

File: xml_converter/generators/generate_cpp.py
Comment (klingbolt) [inline xml_converter/generators/generate_cpp.py:1]: `"functional"` should be added to the list of absolute includes in function `generate_cpp_variable_data`

**Review by klingbolt — CHANGES_REQUESTED** [review]

File: xml_converter/generators/generate_cpp.py [hunk 1]
```diff
@@ -324,7 +324,7 @@ def generate_cpp_variable_data(
             serialize_xml_side_effects: List[str] = []
             deserialize_xml_function: str = "xml_attribute_to_" + class_name
             deserialize_xml_side_effects: List[str] = []
-            serialize_proto_function: str = "to_proto_" + class_name
+            serialize_proto_function: str = class_name + "_to_proto"
             serialize_proto_side_effects: List[str] = []
             deserialize_proto_function: str = "from_proto_" + class_name
```
[the remark this answers is not in the document]
Comment (AsherGlick) [reply xml_converter/generators/generate_cpp.py:329]: This PR only addresses the proto *writing* functions. `"from_proto_" + class_name` is a *reading* function which are changed in #202. This change is already there: https://github.com/AsherGlick/Burrito/pull/202/files#diff-70a30a776b04e988e9b365e1fc13c586b14566b5fec83a08a0096acd249a0ff2R329

If this change were made here then all the *reading* function names would have to be updated too.

File: xml_converter/generators/generate_cpp.py
Comment (AsherGlick) [reply xml_converter/generators/generate_cpp.py:1]: done

File: xml_converter/generators/generate_cpp.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (klingbolt) [reply xml_converter/generators/generate_cpp.py:330]: Makes sense. It can be addressed later.

**Review by klingbolt — APPROVED** [review]

**Pull request merged**